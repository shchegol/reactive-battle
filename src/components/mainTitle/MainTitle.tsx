import React, { FC } from 'react';
import toClassNames from '@root/utils/toClassNames';

import './mainTitle.scss';
import { Props } from './types';

const avatarTmpl = require('@root/images/tank.png').default;

/**
 * Main title
 * @param {string} [titleText='REACTIVE BATTLE'] - title text
 * @param {string} [subtitleText=''] - subtitle text
 * @param [imgSrc=logo] - img file
 * @param {*} rest - rest parameters
 * @constructor
 */

const MainTitle: FC<Props> = ({
  titleText = 'REACTIVE BATTLE',
  subtitleText = '',
  imgSrc = avatarTmpl,
  ...rest
}) => (
  <header
    className={toClassNames(
      'main-title',
      rest.className,
    )}
  >
    <img
      src={imgSrc}
      alt={subtitleText}
      className="main-title__img"
    />

    <h1 className="main-title__text">
      {titleText}
    </h1>

    <p className="main-title__subtitle">
      {subtitleText}
    </p>
  </header>
);

export default MainTitle;
