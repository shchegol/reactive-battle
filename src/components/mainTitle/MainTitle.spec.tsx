import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainTitle from './index';

configure({ adapter: new Adapter() });

describe('MainTitle', () => {
  const snapshoot = (title: string, subtitle: string, imgSrc: string) => {
    const wrapper = render(<MainTitle
      titleText={title}
      subtitleText={subtitle}
      imgSrc={imgSrc}
    />);
    it('Render', () => {
      expect(wrapper).toMatchSnapshot();
    });
  };

  snapshoot('reactive-battle', 'forum', 'https://ya.ru/title.png');
});
