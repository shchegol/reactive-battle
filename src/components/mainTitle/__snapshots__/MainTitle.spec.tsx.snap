// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle Render 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "main-title",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "alt": "forum",
          "class": "main-title__img",
          "src": "https://ya.ru/title.png",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "main-title__text",
          },
          "children": Array [
            Node {
              "data": "reactive-battle",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "main-title__subtitle",
            },
            "children": Array [
              Node {
                "data": "forum",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "class": undefined,
          "src": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "main-title__text",
        },
        "children": Array [
          Node {
            "data": "reactive-battle",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "main-title__subtitle",
          },
          "children": Array [
            Node {
              "data": "forum",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "alt": "forum",
            "class": "main-title__img",
            "src": "https://ya.ru/title.png",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "main-title__subtitle",
        },
        "children": Array [
          Node {
            "data": "forum",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "main-title__text",
          },
          "children": Array [
            Node {
              "data": "reactive-battle",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "alt": "forum",
              "class": "main-title__img",
              "src": "https://ya.ru/title.png",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "class": undefined,
              "src": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "header",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
